activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1548,7 +1548,14 @@ def sanitize_order_arguments(order_args)
       end
 
       def column_references(order_args)
-        references = order_args.grep(String)
+        references = order_args.map do |arg|
+          case arg
+          when String
+            arg
+          when Hash
+            arg.keys
+          end
+        end.flatten
```

Comment:
```suggestion
        references = order_args.flat_map do |arg|
          case arg
          when String
            arg
          when Hash
            arg.keys
          end
        end
```

If any of the hash's keys are arrays, we don't want to flatten them.

Reply:
@eugeneius Thanks!

I've applied the suggestion but the above use case it's unclear to me: could you show me an example of using `Array` as `Hash` key for `order`? I've tried but I couldn't manage to reproduce.

Reply:
I wasn't concerned about breaking a valid use case, more just that `flat_map` corresponds better to what we're trying to do here. Using `flap_map` also saves an intermediate array allocation so [it should be (marginally) faster](https://github.com/rubocop/rubocop-performance/blob/master/lib/rubocop/cop/performance/flat_map.rb), which is probably worthwhile since this code runs every time a relation with an order is built.